iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -522,19 +519,23 @@ private async Task DisposeMessageAsync(string lockToken, AmqpIoTDisposeActions o
 
         private void TwinMessageListener(Twin twin, string correlationId, TwinCollection twinCollection)
         {
-            if (correlationId != null)
+            if (correlationId == null)
```

Comment:
changing logic here to first handle desired property updates and then look at the type of message.